### PR TITLE
CAMEL-14805: Component - Use doInit for wiring instead of doStart

### PR DESCRIPTION
**Pull request by zhfeng**
* camel-hbase
* camel-jing
* camel-jpa
* camel-msv
* camel-pulsar
* camel-snmp
* camel-spark
* camel-stax
* camel-undertow
* camel-xmlsecurity
* camel-xslt-saxon

### CODE REVIEW

File: components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java [hunk 1]
```diff
@@ -265,8 +265,8 @@ public void setProtocol(String protocol) {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
 
         URI uri = URI.create(getEndpointUri());
```
Comment (davsclaus) [inline components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java:271]: Oh there may be a problem when creating an URI as its a network resource. And its either URL or URI in Java that does some networking thingy when being created. @gnodet @ppalaga @lburgazzoli @oscerd @jamesnetherton ?

Comment (gnodet) [reply components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java:271]: No, there is no network access when creating a `URI` afaik.

Comment (oscerd) [reply components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java:271]: I don't think so

Comment (davsclaus) [reply components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpEndpoint.java:271]: Ah thanks yeah I found the answer too in the javadocs, URI mentions further below that URL on contrast does IO networking: https://docs.oracle.com/javase/7/docs/api/java/net/URI.html

**Pull request merged**